spec/services/interaction_tracking_service_spec.rb

```diff
@@ -17,47 +17,40 @@
     let!(:tax_return_2) { create(:tax_return, assigned_user_id: user_no_notifications.id, year: (Rails.configuration.product_year - 1), client: client) }
     let(:user) { create(:admin_user, email_notification: "yes") }
     let(:user_no_notifications) { create(:admin_user, email_notification: "no") }
+    let(:job) { spy("JobSpy") }
 
     before do
       allow(Flipper).to receive(:enabled?).with(:hub_email_notifications).and_return(true)
-      allow(InternalEmail).to receive(:create!).and_call_original
-      allow(SendInternalEmailJob).to receive(:perform_later)
+      allow(ClientInteractionNotificationEmailJob).to receive(:set).and_call_original
+      allow(ClientInteractionNotificationEmailJob).to receive(:set).with(wait: 10.minutes).and_return(job)
+      allow(job).to receive(:perform_later).with(any_args)
+      allow(ClientInteraction).to receive(:create!).and_call_original
     end
 
-    it "sends a notification, enqueues the email job and creates an Internal Email" do
-      described_class.record_incoming_interaction(client, set_flag: true, message_received_at: fake_time)
-      expect(InternalEmail).to have_received(:create!).with(
-        mail_class: UserMailer,
-        mail_method: :incoming_interaction_notification_email,
-        mail_args: ActiveJob::Arguments.serialize(
+    context "client has assigned users, flipper flag hub_email_notifications is enabled and assigned users have opted-in to notifications" do
+      it "sends a notification, enqueues the email job and creates an Internal Email" do
+        described_class.record_incoming_interaction(client, received_at: fake_time, interaction_type: "new_client_message")
+        expect(ClientInteraction).to have_received(:create!).with(
           client: client,
-          user: user,
-          message_received_at: fake_time
+          interaction_type: "new_client_message"
         )
-      )
-      expect(SendInternalEmailJob).to have_received(:perform_later)
-      expect(InternalEmail.last.mail_class).to eq "UserMailer"
-      expect(InternalEmail.last.mail_method).to eq "incoming_interaction_notification_email"
+        expect(ClientInteractionNotificationEmailJob).to have_received(:set).with(wait: 10.minutes)
+        expect(job).to have_received(:perform_later)
+      end
     end
 
-    it "doesn't send a message for the user that has chosen to opt-out of email notifications" do
-      described_class.record_incoming_interaction(client, set_flag: true, message_received_at: fake_time)
-      expect(InternalEmail).not_to have_received(:create!).with(
-        mail_class: UserMailer,
-        mail_method: :incoming_interaction_notification_email,
-        mail_args: ActiveJob::Arguments.serialize(
-          client: client,
-          user: user_no_notifications,
-          message_received_at: fake_time
-        )
-      )
+    context "user that has chosen to opt-out of email notifications" do
```

Comment:
where did this context define the `opt-out of email notifications` part of the test set up?